src/arviz_plots/plots/convergencedistplot.py

```diff
@@ -0,0 +1,202 @@
+
+    Returns
+    -------
+    PlotCollection
+
+    Examples
+    --------
+    Select a single variable and specify diagnostics
+
+    .. plot::
+        :context: close-figs
+
+        >>> from arviz_plots import plot_convergence_dist, style
+        >>> style.use("arviz-clean")
+        >>> from arviz_base import load_arviz_data
+        >>> rugby = load_arviz_data('radon')
+        >>> plot_convergence_dist(radon, var_names=["za_county"], diagnostics=["rhat", "ess_tail"])
+
+     Some ess methods accepts a probability argument
+
+    .. plot::
+        :context: close-figs
+
+        >>> plot_convergence_dist(radon, var_names=["za_county"],
+                     diagnostics=["ess_tail(0.1, 0.9)",
+                                  "ess_local(0.1, 0.9)",
+                                  "ess_quantile(0.9)"])
+
+
+    .. minigallery:: plot_convergence_dist
+
+    """
+    if sample_dims is None:
+        sample_dims = rcParams["data.sample_dims"]
+    if isinstance(sample_dims, str):
+        sample_dims = [sample_dims]
+    sample_dims = list(sample_dims)
+    if plot_kwargs is None:
+        plot_kwargs = {}
+    else:
+        plot_kwargs = plot_kwargs.copy()
+    if pc_kwargs is None:
+        pc_kwargs = {}
+    else:
+        pc_kwargs = pc_kwargs.copy()
+
+    if diagnostics is None:
+        diagnostics = ["ess_bulk", "ess_tail", "rhat_rank"]
+
+    dt = process_group_variables_coords(
+        dt, group=group, var_names=var_names, filter_vars=filter_vars, coords=coords
+    )
+
+    new_ds = _get_diagnostics(dt, diagnostics)
+
+    plot_kwargs.setdefault("credible_interval", False)
+    plot_kwargs.setdefault("point_estimate", False)
+    plot_kwargs.setdefault("point_estimate_text", False)
+
+    plot_collection = plot_dist(
+        new_ds,
+        var_names=None,
+        filter_vars=None,
+        group=None,
+        coords=None,
+        sample_dims=sample_dims,
+        kind=kind,
+        point_estimate=point_estimate,
+        ci_kind=ci_kind,
+        ci_prob=ci_prob,
+        plot_collection=plot_collection,
+        backend=backend,
+        labeller=labeller,
+        aes_map=aes_map,
+        plot_kwargs=plot_kwargs,
+        stats_kwargs=stats_kwargs,
+        pc_kwargs=pc_kwargs,
+    )
+
+    if ref_line:
+        plot_kwargs.setdefault("ref_line", {})
+        if plot_kwargs["ref_line"] is not False:
+            plot_kwargs["ref_line"].setdefault("color", "k")
+            plot_kwargs["ref_line"].setdefault("linestyle", "--")
+            plot_kwargs["ref_line"].setdefault("alpha", 0.5)
+
+        ess_ref = dt.sizes["chain"] * 100
+        # is this valid for all r_hat methods? Do we want to correct for multiple comparisons?
+        r_hat_ref = 1.01
+        for diagnostic in diagnostics:
+            if "ess" in diagnostic:
+                ref = ess_ref
+            if "rhat" in diagnostic:
+                ref = r_hat_ref
+            plot_collection.viz[diagnostic]["plot"].item().axvline(
```

Comment:
This only works with Matplotlib as other backends not necessarily have the `axvline` method

Reply:
yep, plotting backends should not be called directly but via thin wrappers that act as translators so our plotting functions see a common api independently of the backend being used. It also needs tests before it can be merged, I'll try to push it forward a bit more now